code/game/objects/items/stacks/stack.dm

```diff
@@ -27,6 +27,8 @@
 	var/merge_type = null // This path and its children should merge with this stack, defaults to src.type
 	var/recipe_width = 400 //Width of the recipe popup
 	var/recipe_height = 400 //Height of the recipe popup
+	/// What sort of table is made when applying this stack to a frame?
+	var/table_type
```

Comment:
I'm not sure I find this logical. This makes stacks tightly coupled to tables.
I think a better solution would be to have a mapping in the `table_frame` object which basically says which stack corresponds with which table type.
metal -> metal table
plasma -> plasma table
etc

This way the stack has no clue about tables. And the tables control how they get made

For the edge cases, such as `table_frame/abductor` and `table_frame/wood` you can just override the behaviour

Reply:
I think stacks should be tightly coupled to tables, considering that is how you make tables. I can make a giant assoc list if you'd like and loop through it every time someone hits a table frame, but a single `stack?.table_type` check seems incredibly simpler and more efficient. Not to mention we already do this for window types using the `created_window` var on stacks.